Apps/WebClient/src/ClientApp/src/components/timeline/TimelineComponent.vue

```diff
@@ -503,19 +509,70 @@ function fetchDataset(entryType: EntryType): Promise<void> {
 }
 
 function fetchTimelineData(): Promise<void | void[]> {
-    const promises = props.entryTypes.map((entryType) =>
-        fetchDataset(entryType)
-    );
+    const cannotAccessEntryTypes: EntryType[] = [];
+
+    const promises: Promise<void>[] = [];
+    props.entryTypes.forEach((entryType) => {
+        {
+            if (canAccessDataset(entryType)) {
+                promises.push(fetchDataset(entryType));
+            } else {
+                cannotAccessEntryTypes.push(entryType);
+            }
+        }
+    });
 
     if (props.commentsAreEnabled) {
         promises.push(retrieveComments(props.hdid));
     }
 
+    showDatasetWarning(cannotAccessEntryTypes);
+
     return Promise.all(promises).catch((err) =>
         logger.error(`Error loading timeline data: ${err}`)
     );
 }
 
+function canAccessDataset(entryType: EntryType): boolean {
+    const dataSource = dataSourceUtil.getDataSource(entryType);
+    const canAccess = !blockedDataSources.value.includes(dataSource);
+    logger.debug(`Can access data source for ${dataSource}: ${canAccess}`);
+    return canAccess;
+}
+
+function showDatasetWarning(entryTypes: EntryType[]): void {
+    const entryTypesLength = entryTypes.length;
+    logger.debug(
+        `Cannot access dataset entry types length: ${entryTypesLength}`
+    );
+
+    if (entryTypesLength > 0) {
```

Comment:
Nit: I'd suggest instead using an if block that checks `entryTypes.length === 0` and returns early, to reduce indentation.

Reply:
Changed